### ISSUE TITLE
Quarto beamer doesn't generate content when I set `toc: true`

### ISSUE THREAD

**Issue by XiangyunHuang** [quarto-dev/quarto-cli#6681]
### Bug description

Quarto beamer doesn't generate content when I set `toc: true`

### Steps to reproduce

Here is a minimal qmd file.

````qmd
---
title: "Quarto"
date: today
date-format: long
documentclass: beamer
format:
  beamer:
    pdf-engine: lualatex
    keep-tex: true
    toc: true
---

# In the morning {#sec-mo}

## Getting up

-   Turn off alarm
-   Get out of bed

## Breakfast

-   Eat eggs
-   Drink coffee

# In the evening {#sec-ev}

## Dinner

-   Eat spaghetti
-   Drink wine

## Going to sleep

-   Get in bed
-   Count sheep
````

### Expected behavior

<img width="963" alt="截屏2023-09-02 10 44 43" src="https://github.com/quarto-dev/quarto-cli/assets/12031874/4b4af68c-e1c8-4820-91a5-78f3577a9b63">

### Actual behavior

<img width="958" alt="截屏2023-09-02 10 45 52" src="https://github.com/quarto-dev/quarto-cli/assets/12031874/13639d55-4515-4b86-8086-0f429d506a3b">

### Your environment

- IDE: RStudio 2023.06.2+561 "Mountain Hydrangea" Release (de44a3118f7963972e24a78b7a1ad48b4be8a217, 2023-08-22) for macOS
- OS: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.06.2+561 Chrome/110.0.5481.208 Electron/23.3.0 Safari/537.36

### Quarto check output

```bash
quarto check
```
```
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.450
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.5
      Path: /usr/local/opt/python@3.11/bin/python3.11
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.3.1
      Path: /usr/local/Cellar/r/4.3.1/lib/R
      LibPaths:
        - /usr/local/lib/R/4.3/site-library
        - /usr/local/Cellar/r/4.3.1/lib/R/library
      knitr: 1.43
      rmarkdown: 2.24

[✓] Checking Knitr engine render......OK
```

**Comment by XiangyunHuang** [comment]
Here is output from Background Jobs window.

````

pandoc 
  to: beamer
  output-file: quarto-test.tex
  standalone: true
  pdf-engine: lualatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  toc: true
  
metadata
  block-headings: true
  title: Quarto
  date: today
  date-format: long
  documentclass: beamer
  
running lualatex - 1
  This is LuaHBTeX, Version 1.17.0 (TeX Live 2023) 
   restricted system commands enabled.
  

Output created: quarto-test.pdf
````

I guess the root cause is that we only run lualatex once.

**Comment by XiangyunHuang** [comment]
When I set `latex-min-runs: 2`, the problem is solved.

**Comment by cscheid** [comment]
We use latexmk to determine the number of passes through the document, and I think it's failing in this scenario.

@dragonstyle we should be able to set `latex-min-runs` given the presence of beamer format and toc: true, right?

**Comment by cderv** [comment]
@XiangyunHuang can you try a pre-release version of Quarto ? https://quarto.org/docs/download/prerelease.html

With 1.3 I can see the issue, but I think it is fixed already as I can't using 1.4 version.

So I cannot reproduce this with current 1.4 version (soon to be release) and I get a TOC from your example above (https://github.com/quarto-dev/quarto-cli/issues/6681#issue-1878291139)

**Comment by XiangyunHuang** [comment]
@cderv I confirm that this issue is solved by Quarto 1.4. And I just find out a similar issue https://github.com/quarto-dev/quarto-cli/issues/6226

**Comment by cderv** [comment]
Duplicate of #6226
